camel/models/schema_model.py

```diff
@@ -0,0 +1,233 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+import json
+from typing import (
+    Any,
+    Dict,
+    List,
+    Optional,
+    Type,
+    TypeVar,
+    Union,
+    overload,
+)
+
+from openai import Stream
+from pydantic import BaseModel, ValidationError
+
+from camel.messages import OpenAIMessage
+from camel.models import BaseModelBackend
+from camel.types import (
+    ChatCompletion,
+    ChatCompletionChunk,
+    ChatCompletionMessage,
+    Choice,
+    ModelPlatformType,
+    ModelType,
+)
+from camel.utils import (
+    BaseTokenCounter,
+    OpenAITokenCounter,
+)
+
+T = TypeVar('T', bound=BaseModel)
+
+
+class SchemaModel(BaseModelBackend):
+    r"""Shema model in a unified BaseModelBackend interface, which aims to
+    generate the formatted response."""
+
+    def __init__(
+        self,
+        model_platform: ModelPlatformType,
+        model_type: str,
+        model_config_dict: Dict[str, Any],
+        url: Optional[str] = None,
+    ) -> None:
+        r"""Constructor for OpenAI backend.
+
+        Args:
+            model_platform (ModelPlatformType): Platform from which the model
+                originates, including transformers, llama_cpp, and vllm.
+            model_type (str): Model for which a backend is created, for
+                example, "mistralai/Mistral-7B-v0.3".
+            model_config_dict (Dict[str, Any]): A dictionary that will
+                be fed into openai.ChatCompletion.create().
+            url (Optional[str]): The url to the OpenAI service.
+        """
+        from outlines import models  # type: ignore[import]
```

Comment:
1. The package `outlines` is added into `pyproject.toml` but is not added into `tools` and `all` array, therefore when users are running `poetry install -E all --with dev,docs`, `outlines` would not be installed. 
2. If the `# type: ignore[import]` is used to let `mypy` skip the missing import type stubs, I guess we can add `outlines.*` into the last section of `pyproject.toml` directly?

Reply:
outlines is added if the option is 'all', however ci/cd does not install outlines (and it should better not install it). In order to pass the mypy, I have to use type: ignore.